node/node.go

```diff
@@ -122,6 +218,7 @@ func NewNode(config *cfg.Config,
 	clientCreator proxy.ClientCreator,
 	genesisDocProvider GenesisDocProvider,
 	dbProvider DBProvider,
+	metricsProvider MetricsProvider,
```

Comment:
`NewNode` already takes the config, no need to extend its surface even further we can do the setup of what metric to use for what component down below.